lib/http.js

```diff
@@ -239,6 +240,31 @@ module.exports = function (chai, _) {
     .keys(contentTypes)
     .forEach(checkContentType);
 
+  /**
+   * ### .charset
+   *
+   * Assert that a `Response` or `Request` object has a given charset.
+   *
+   * ```js
+   * expect(req).to.have.charset('utf-8');
+   * ```
+   *
+   * @name charset
+   * @api public
+   */
+
+  Assertion.addMethod('charset', function (value) {
+    var headers = this._obj.headers;
+    var cs = charset(headers);
+    var expected = value.replace("-", "");
```

Comment:
Is this to transform `'utf-8'` to `'utf8'`? If so, I'm thinking that might be too specific a case. What do you think about using @keithamus's suggestion of [`node-mime`](https://github.com/broofa/node-mime) instead? (See his comment here https://github.com/chaijs/chai-http/issues/26#issuecomment-69003983) It seems to be better maintained and documented as well.

Reply:
Thanks @austince. I did look the suggestion to use `node-mime` however as of version 2 `charset()` has been removed which is why I went with the `charset` module.

You do need to remove the '-' in order for the comparison to succeed. An improvement might be to convert `value` to all lowercase first. That satisfies input like:
- "UTF-8"
- "utf-8"
- "UTF8"
- "utf8"

and variants.

Reply:
Ah, that's unfortunate about `node-mime`, but makes sense. 

It looks like from the `charset` code that `utf-8` is a special case:

https://github.com/node-modules/charset/blob/dcc48ee609a9b8a137c68faad712c646fcaeda29/index.js#L54-L56

If we're going to use this module, do you think it might be better to handle it as a special case on our end as well, instead of removing the `-` from all inputs? 

I think lowercasing the input value sounds like a good idea!

Reply:
Good catch there @austince.